Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -582,6 +582,13 @@ private void RetestSubmission(DatabaseModelType submission)
                 return;
             }
 
+            var submissionForProcessing = this.Data.SubmissionsForProcessing.GetById(submission.Id);
+
+            if (submissionForProcessing != null && !submissionForProcessing.Processed)
```

Comment:
Is this check sufficient to determine whether the _actual_ submission has been `Processed` ? What if it  isn't present in the SubmissionForProcessing table?